pino/index.d.ts

```diff
@@ -10,9 +10,18 @@ import stream = require('stream');
 declare function P(optionsOrStream?: P.LoggerOptions | stream.Writable | stream.Duplex | stream.Transform): P.Logger;
 declare function P(options: P.LoggerOptions, stream: stream.Writable | stream.Duplex | stream.Transform): P.Logger;
 
+interface IPinoLog {
+    pid: number
+    hostname: string
+    level: number
+    time: string
+    msg: string
+    v: number
+}
+
 declare namespace P {
 
-    function pretty(opts?: {timeTransOnly?: boolean, levelFirst?: boolean, formatter?: (str: string) => void}): stream.Transform;
+    function pretty(opts?: {timeTransOnly?: boolean, levelFirst?: boolean, formatter?: (log: IPinoLog) => string}): stream.Transform;
```

Comment:
I think the IPinoLog interface isn't quite correct, and I don't think it is required.
When I run pretty(), and JSON.stringify the passed object, I find:
> time: number

I still prefer the simple typing of **{}** as in my suggestion.

Also keep in mind that there will be many more fields in the object, that is, all of those that are dynamically added from the type to be logged.

@DanielRosenwasser 
There used to be a guideline for writing declarations that advised against typing parts of the interface that aren't documented, but I can no longer find this.
The type of the argument to *formatter* is only documented as an object, or **{}**